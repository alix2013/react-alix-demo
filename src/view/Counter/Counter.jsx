import React from "react";
import { BaseComponent } from "react-alix";

export default class Counter extends BaseComponent {
  store = this.props.store.counterStore;

  componentDidMount() {
    super.componentDidMount();
    this.dispatch(this.store, "monitorState");
  }

  componentWillUnmount() {
    this.dispatch(this.store, "unMonitorState");
  }
  render() {
    this.debug("Store in render:", this.store, this.store.count);
    console.debug();
    // this.debug("line:", /\(file:[\w\d/.-]+:([\d]+)/.exec(new Error().stack)[0]);
    return (
      <div className="title">
        <h2> Counter Demo ( View - ViewModel ) </h2>
        {/* <input ref={c => (this._input = c)} /> */}
        <div className="counter">
          <button
            className="counter-button"
            type="button"
            id="incrButton"
            onClick={() => this.dispatch(this.store, "increaseAction")}
          >
            +
          </button>
          <span id="countValue">{this.store.count}</span>
          <button
            className="counter-button"
            id="decreseButton"
            type="button"
            onClick={() => this.dispatch(this.store, "decreaseAction")}
          >
            -
          </button>
        </div>
      </div>
    );
  }
}
