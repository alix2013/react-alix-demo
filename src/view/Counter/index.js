import { injectObsever } from "react-alix";
import component from "./Counter";

export default injectObsever("store", component);
