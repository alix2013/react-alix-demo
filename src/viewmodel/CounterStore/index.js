import { decorate, observable, action } from "react-alix";

import CounterStore from "./CounterStore";

decorate(CounterStore, {
  count: observable,
  increaseAction: action,
  decreaseAction: action
});

export default new CounterStore();
