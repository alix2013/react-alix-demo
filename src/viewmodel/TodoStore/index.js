import { decorate, observable, action } from "react-alix";
import TodoStore from "./TodoStore";
decorate(TodoStore, {
  isLoading: observable,
  currentPage: observable,
  newInputValue: observable,
  errorMessage: observable,
  dataList: observable,
  isEditMode: observable,
  //   queryDataAction: action,
  //   queryDataActionV2: action,
  setIsLoadingAction: action,
  setDataListAction: action,
  setErrorMessageAction: action,
  setNewValue: action,
  addAction: action,
  prePageAction: action,
  nextPageAction: action,
  deleteAction: action,
  completeAction: action,
  setEditModeAction: action,
  editAction: action
});

export default new TodoStore();
