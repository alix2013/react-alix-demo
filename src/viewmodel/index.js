import todoStore from "./TodoStore";
import counterStore from "./CounterStore";
// const store = {
//   todoStore,
//   counterStore
// };
// export default store;
export { todoStore, counterStore };
