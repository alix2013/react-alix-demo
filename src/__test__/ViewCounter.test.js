import "babel-polyfill";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import { shallow, mount } from "enzyme";
import Counter from "../view/Counter";
// import store from "../viewmodel";
import { counterStore } from "../viewmodel";
configure({ adapter: new Adapter() });

// import { Store } from "react-alix";
// const store = new Store().add("counterStore", counterStore);
const store = { counterStore };

describe("Test counter", () => {
  it("view shallow test", () => {
    console.log("testing shallow...");

    const wrapper = shallow(<Counter store={store} />);
    console.log("debug=>", wrapper.debug());
    // console.log("instance=>", wrapper.instance());

    // console.log("instance=>", wrapper.props());
    // console.log("instance=>", wrapper.prop("store"));

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.instance().props.store).toBe(store);
  });
});

describe("Test counter", () => {
  it("view test", () => {
    const comp = mount(<Counter store={store} />);
    console.log("testing mount...");
    expect(comp.props().store.counterStore.count).toBe(0);
    let button = comp.find("#incrButton");
    button.simulate("click");
    // console.log(button.props());
    //trigger click by method onClick()
    button.props().onClick();

    console.log("current value->", comp.find("#countValue").text());
    expect(comp.props().store.counterStore.count).toBe(2);

    comp.find("#incrButton").simulate("click");
    console.log("current value->", comp.find("#countValue").text());
    expect(comp.find("#countValue").text()).toBe("3");
    expect(comp.props().store.counterStore.count).toBe(3);

    comp.find("#decreseButton").simulate("click");
    expect(comp.props().store.counterStore.count).toBe(2);
    expect(comp.find("#countValue").text()).toBe("2");
    console.log("current value->", comp.find("#countValue").text());
  });
});
