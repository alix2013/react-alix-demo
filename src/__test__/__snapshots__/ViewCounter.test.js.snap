// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test counter view shallow test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <inject-Counter-with-store
    store={
      Object {
        "counterStore": CounterStore {
          "addService": [Function],
          "cancel": [Function],
          "cancelAll": [Function],
          "count": 0,
          "debug": [Function],
          "error": [Function],
          "getService": [Function],
          "removeService": [Function],
          "services": Array [],
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "CounterStore@2.keys",
              "observers": Array [],
            },
            "name": "CounterStore@2",
            "target": [Circular],
            "values": Array [
              Array [
                "count",
                0,
              ],
            ],
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "store": Object {
        "counterStore": CounterStore {
          "addService": [Function],
          "cancel": [Function],
          "cancelAll": [Function],
          "count": 0,
          "debug": [Function],
          "error": [Function],
          "getService": [Function],
          "removeService": [Function],
          "services": Array [],
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "CounterStore@2.keys",
              "observers": Array [],
            },
            "name": "CounterStore@2",
            "target": [Circular],
            "values": Array [
              Array [
                "count",
                0,
              ],
            ],
          },
        },
      },
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "store": Object {
          "counterStore": CounterStore {
            "addService": [Function],
            "cancel": [Function],
            "cancelAll": [Function],
            "count": 0,
            "debug": [Function],
            "error": [Function],
            "getService": [Function],
            "removeService": [Function],
            "services": Array [],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "CounterStore@2.keys",
                "observers": Array [],
              },
              "name": "CounterStore@2",
              "target": [Circular],
              "values": Array [
                Array [
                  "count",
                  0,
                ],
              ],
            },
          },
        },
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
